<launch>

  <arg name="init_x" value="-3.0" />
  <arg name="init_y" value="0.0" />
  <arg name="init_z" value="1.0" />
  <arg name="map_size_x" value="15" />
  <arg name="map_size_y" value="15" />
  <arg name="map_size_z" value="5" />
  <arg name="pool_size_x" value="120" />
  <arg name="pool_size_y" value="120" />
  <arg name="pool_size_z" value="50" />
  <arg name="cx" value="321.04638671875" />
  <arg name="cy" value="243.44969177246094" />
  <arg name="fx" value="387.229248046875" />
  <arg name="fy" value="387.229248046875" />


  <node pkg="plan_env" name="risk_voxel" type="risk_voxel_node" output="screen">
    <param name="init_x" value="$(arg init_x)" />
    <param name="init_y" value="$(arg init_y)" />
    <param name="init_z" value="$(arg init_z)" />
    <param name="vel" value="1" />
    <param name="time_step" value="0.01" />

    <param name="map/local_update_range_x" value="5.5" />
    <param name="map/local_update_range_y" value="5.5" />
    <param name="map/local_update_range_z" value="4.5" />

    <!-- Booleans -->
    <param name="map/booleans/pub_world_frame" value="true" />
    <param name="map/booleans/pub_spatio_temporal" value="false" />

    <!-- Fake Perception -->
    <remap from="~map/state" to="/ground_truth_state" />
    <remap from="~map/odom" to="/odom" />
    <remap from="~map/cloud" to="/local_cloud" />
    <remap from="~map/pose" to="/drone_camera_pose_topic" />
    <remap from="~map/depth" to="/drone_depth_topic" />
    <!-- Resolutions -->
    <param name="map/resolution" value="0.2" />
    <param name="map/angle_resolution" value="3" />
    <param name="map/time_resolution" value="0.2" />
    <param name="map/n_risk_map" value="3" />
    <param name="map/n_prediction_per_risk" value="3" />
    <param name="map/n_particles_max_per_voxel" value="18" />
    <!-- Variances -->
    <param name="map/stddev_pos" value="0.05" />
    <param name="map/stddev_vel" value="0.05" />
    <param name="map/sigma_update" value="0.1" />
    <param name="map/sigma_observation" value="0.05" />
    <param name="map/sigma_localization" value="0.00" />
    <!-- Newborn particles -->
    <param name="map/newborn/particles_weight" value="0.0001" />
    <param name="map/newborn/particles_number" value="20" />
    <param name="map/newborn/objects_weight" value="0.04" />
    <!-- FOV settings -->
    <param name="map/half_fov_horizontal" value="42" />
    <param name="map/half_fov_vertical" value="24" />

    <!-- Useless in DSP Map -->
    <param name="map/virtual_ceil_height" value="2.9" />
    <param name="map/visualization_truncate_height" value="1.8" />
    <param name="map/show_occ_time" value="false" />
    <param name="map/pose_type" value="1" />
    <param name="map/frame_id" value="world" />
    <param name="map/risk_threshold" value="0.2" />
  </node>

  <node pkg="map_generator" name="map_generator" type="dynamic_forest_seq">
    <remap from="~global_cloud" to="/map_generator/global_cloud" />
    <remap from="~global_cylinder_state" to="/ground_truth_state" />
    <param name="init_state_x" value="0.0" />
    <param name="init_state_y" value="0.0" />
    <param name="map/x_size" value="$(arg map_size_x)" />
    <param name="map/y_size" value="$(arg map_size_y)" />
    <param name="map/z_size" value="$(arg map_size_z)" />
    <param name="map/resolution" value="0.10" />
    <param name="map/frame_id" value="world" />
    <param name="ObstacleShape/seed" value="-1" />
    <param name="map/obs_num" value="20" />
    <param name="ObstacleShape/lower_rad" value="0.1" />
    <param name="ObstacleShape/upper_rad" value="0.5" />
    <param name="ObstacleShape/lower_hei" value="4.0" />
    <param name="ObstacleShape/upper_hei" value="4.0" />
    <param name="ObstacleShape/upper_vel" value="0.05" />
    <param name="ObstacleShape/set_cylinder" value="true" />
    <param name="sensing/radius" value="5.0" />
    <param name="sensing/rate" value="10.0" />
  </node>

  <node pkg="poscmd_2_odom" name="poscmd_2_odom" type="poscmd_2_odom" output="screen">
    <param name="init_x" value="$(arg init_x)" />
    <param name="init_y" value="$(arg init_y)" />
    <param name="init_z" value="$(arg init_z)" />
    <remap from="~command" to="/pos_command" />
    <remap from="~odometry" to="/odom" />
  </node>

  <!-- Disable for Fake Perception -->
  <node pkg="local_sensing_node" type="pcl_render_node" name="pcl_render_node">
    <rosparam command="load" file="$(find local_sensing_node)/params/camera.yaml" />
    <param name="sensing_horizon" value="5.0" />
    <param name="sensing_rate" value="30.0" />
    <param name="estimation_rate" value="30.0" />
    <param name="is_camera_frame" value="true" />
    <param name="is_dynamic_map" value="true" />

    <param name="map/x_size" value="$(arg map_size_x)" />
    <param name="map/y_size" value="$(arg map_size_y)" />
    <param name="map/z_size" value="$(arg map_size_z)" />

    <remap from="~global_map" to="/map_generator/global_cloud" />
    <remap from="~odometry" to="/odom" />
    <remap from="~pcl_render_node/cloud" to="/local_cloud" />
  </node>

  <node pkg="rviz" name="rviz" type="rviz" args="-d $(find plan_env)/rviz/dynamic_map.rviz" />

</launch>
